src/popup/functions.js

```diff
@@ -81,27 +105,38 @@ export function deleteIntervals() {
 
 export function checkLanguageSupport(api, tab, setLanguageSupported) {
   Article(tab.url).then((article) => {
-    api.isArticleLanguageSupported(
-      article.textContent,
-      (result_dict) => {
-        console.log(result_dict);
-        if (result_dict === "NO") {
-          setLanguageSupported(false);
-        }
-        if (result_dict === "YES") {
-          setLanguageSupported(true);
-        }
+    api.isArticleLanguageSupported(article.textContent, (result_dict) => {
+      console.log(result_dict);
+      if (result_dict === "NO") {
+        setLanguageSupported(false);
       }
-    );
+      if (result_dict === "YES") {
+        setLanguageSupported(true);
+      }
+    });
+  });
+}
+
+export function checkLanguageSupportFromUrl(api, url, setLanguageSupported) {
+  Article(url).then((article) => {
```

Comment:
This was a way to call the same as we do in the PopUpContent, but you are right we can simplify it. Though in the case of the PopUp content I think calling by URL rather than tab might be more generic. 

For our case we can just make a call to the api with the text context.